dashboard/dashboard/middleware.py

```diff
@@ -40,28 +42,40 @@ def __call__(self, request):
             if auth_token is None:
                 logging.error("UserCookieMiddleware: user_auth from session is empty")
                 return response
-
-            # if cookie value contains /, it would get unnecessarily double quoted, so encode
-            value = quote(auth_token, safe="")
-
-            # Domain defines the host to which the cookie will be sent.
-            # Empty domain means that cookie will be created for current host, not including subdomains.
-            # if specified, then subdomains are included.
-            response.set_cookie(
-                key=RT_AUTH_HEADER,
-                value=value,
-                path='/',
-                domain=cookie_domain,
-                max_age=60 * 60 * 24 * 365,  # one year of cookie lifetime (in seconds), also makes cookie persist between browser restarts
-                samesite='lax',  # cookie is sent only when user access origin site or navigates to it, prevents CSRF
-                httponly=True,  # prevents client-side JavaScript read access
-                secure=True)  # cookie is sent only when request is made with https (except on localhost)
+        
+            set_auth_token_cookie(auth_token, response)
 
         elif not user_authenticated:
             if request.COOKIES.get(RT_AUTH_HEADER):
                 response.delete_cookie(
                     key=RT_AUTH_HEADER,
                     path='/',
-                    domain=cookie_domain)
+                    domain=_get_cookie_domain())
 
         return response
+
+
+def set_auth_token_cookie(auth_token: str, response: HttpResponse):
+    # if cookie value contains /, it would get unnecessarily double quoted, so encode
+    value = quote(auth_token, safe="")
+
+    # Domain defines the host to which the cookie will be sent.
+    # Empty domain means that cookie will be created for current host, not including subdomains.
+    # if specified, then subdomains are included.
+    response.set_cookie(
+        key=RT_AUTH_HEADER,
+        value=value,
+        path='/',
+        domain=_get_cookie_domain(),
+        max_age=60 * 60 * 24 * 365,  # one year of cookie lifetime (in seconds), also makes cookie persist between browser restarts
```

Comment:
Is there a reason this lasts an entire year? I don't think it's an issue, it just seems like quite a long time intuitively.

Reply:
Is this cookie age? FWIW I routinely set that to ten years in my own development projects. I think it's pretty common practise.

Reply:
Yes, it's the cookie age. I also think a year is quite a long time. Do we want to decrease it to a month?

Reply:
I think a month makes more sense, but just chose whatever makes sense to you. The exact value isn't super important, a year just seemed odd.

Reply:
So I'm gonna change it to month. @JosefAssadERST  Any objections?

Reply:
No objections, but I'm curious. What ia gained by going from 1 year to 1 month?

Reply:
I guess that's just a good practice in terms of security. If an attacker gains access to a user's computer, they can steal the cookies and impersonate the user. Setting a short lifetime for cookies limits the time frame in which an attacker can use stolen cookies, reducing the risk of a successful attack.

Plus, we'll make sure users remember their password :wink: